src/utils.rs

```diff
@@ -48,6 +48,15 @@ pub fn read_file<P: AsRef<Path> + Debug>(file_name: P) -> Result<String> {
     result
 }
 
+/// Returns the string contents of a file from the current working directory
+pub fn read_file_from_pwd(context: &Context, file_name: &str) -> Option<String> {
+    if !context.try_begin_scan()?.set_files(&[file_name]).is_match() {
+        return None;
+    }
+
+    read_file(context.current_dir.join(file_name)).ok()
+}
```

Comment:
Might be a bit nice to put this into `Context` and have a log message when skipping a file.

```suggestion
impl Context {
    pub fn read_file_from_pwd(&self, file_name: &str) -> Option<String> {
        if !self.try_begin_scan()?.set_files(&[file_name]).is_match() {
            log::debug!("Not attempting to read {file_name} because, it was not included in scan.");
            return None;
        }
    
        read_file(context.current_dir.join(file_name)).ok()
    }
}
```

Reply:
I didn't think about that but now you have mentioned it, it seems so obvious.